snowfakery_samples/npsp/Opportunity_npsp_soft_credits.yml

```diff
@@ -0,0 +1,59 @@
+## NPSP Opportunities with contact soft credits
+
+## create several contacts 
+  - object: Contact
+    count: 5
+    fields:
+      Firstname:
+        fake: first_name
+      Lastname:
+        fake: last_name
+      npe01__AlternateEmail__c:
+        fake: email
+      npe01__HomeEmail__c:
+        fake: email
+      npe01__WorkEmail__c:
+        fake: email
+      npe01__WorkPhone__c:
+        fake: phone_number
+    
+## create an account
+  - object: Account
+    fields:
+      name:
+        fake: company
+    
+##create opportunity for that account
+  - object: Opportunity
+    nickname: GrantOpportunity
+    fields:
+      Name: Grant Opportunity
+      Description: A Grant Opportunity
+      AccountId:
+        reference: Account
+      Amount:
+        random_number:
+          min: 1
+          max: 100000
+      CloseDate: 2021-10-10
+      StageName: 
+        random_choice:
+          Prospecting: 50%
+          Qualification: 50%
+
+## create opportunity contact roles for above opportunity and contacts created earlier
+  - object: OpportunityContactRole
+    count: 5
+    fields:
+      OpportunityId:
+        reference: Opportunity
+      ContactId:
+        random_reference: Contact
+      Role:
+        random_choice:
+          Influencer: 20%
+          Solicitor: 20%
+          Decision Maker: 20%
+          Honoree: 20%
+          Donor: 20%
```

Comment:
@saranya-selvaraj I tested out your changes (finally!) in a scratch org today and it worked great! My only minor suggestion would be to remove the Donor Contact Role. It looks like Aaron actually merged your changes already, so no rush or worries to update any time soon. Thank you and have a great weekend!